File: go/metric_usage.go
// this file is @generated
package meteroid

type MetricUsage struct {
	GroupedUsage RequiredSlice[GroupedUsage] `json:"grouped_usage"`

	MetricCode string `json:"metric_code"`

	MetricId BillableMetricId `json:"metric_id"`

	MetricName string `json:"metric_name"`

	TotalValue string `json:"total_value"`
}

File: python/meteroid/models/metric_usage.py
# this file is @generated
import dataclasses
import typing as t
from decimal import Decimal

from ..serialization import BaseModel
from .billable_metric_id import BillableMetricId
from .grouped_usage import GroupedUsage


@dataclasses.dataclass
class MetricUsage(BaseModel):
    grouped_usage: t.List[GroupedUsage]

    metric_code: str

    metric_id: BillableMetricId

    metric_name: str

    total_value: Decimal
